core/cmd/client.go

```diff
@@ -270,6 +270,25 @@ func (n ChainlinkRunner) Run(ctx context.Context, app chainlink.Application) err
 	return errors.WithStack(g.Wait())
 }
 
+func tryRunServerUntilCancelled(ctx context.Context, lggr logger.Logger, cfg config.GeneralConfig, runServer func() error) {
+	for {
+		// try calling runServer() and log error if any
+		if err := runServer(); err != nil {
+			if err != http.ErrServerClosed {
+				lggr.Criticalf("%v", err)
+			}
```

Comment:
Can ErrServerClosed ever happen unexpectedly and signify any breakage? Or is it only on clean shutdown?

Reply:
in my experience this error is returned from both ListenAndServe() and Shutdown()

Reply:
Also, based on the spec:
```
// ErrServerClosed is returned by the Server's Serve, ServeTLS, ListenAndServe,
// and ListenAndServeTLS methods after a call to Shutdown or Close.
```
it should not happen in other cases

Reply:
Are the logs in `run()` and `runTLS()` redundant now that we log here?
Should this log have a message prefix before the error?

Reply:
(changing in the open PR #5929 too)

Reply:
Fixed